packages/ckeditor5-engine/src/model/schema.ts

```diff
@@ -576,16 +628,15 @@ export default class Schema extends ObservableMixin() {
 	 * {@link module:engine/model/schema~SchemaContext} (context) instance and attribute name.
 	 * The callback may return `true/false` to override `checkAttribute()`'s return value. If it does not return
 	 * a boolean value, the default algorithm (or other callbacks) will define `checkAttribute()`'s return value.
+	 * @param forNode Optional name of the node to run the callback for.
+	 * 	If no node is specified, the callback will run for each `checkAttribute` call.
 	 */
-	public addAttributeCheck( callback: SchemaAttributeCheckCallback ): void {
-		this.on<SchemaCheckAttributeEvent>( 'checkAttribute', ( evt, [ ctx, attributeName ] ) => {
-			const retValue = callback( ctx, attributeName );
+	public addAttributeCheck( callback: SchemaAttributeCheckCallback, forNode?: string ): void {
```

Comment:
Although I think I understand why you went this way. We have two types of callbacks when it comes to current `addAttributeCheck`.

First -- we have custom callbacks for attribute X. They would be solved if you implemented it accordingly to my idea (set callback for attribute name, not element name).

But then, we have attribute-related callbacks for element Y. For example, disallow all attributes in `$text` inside `title-content`. But in this case, such callback should be actually called for all attributes, so it makes sense that it would stay as a generic callback. It's the same as adding one callback for all attributes.

OTOH, if a callback stays generic it will be also called for non-related items. Let's take example rule:

```javascript
schema.addAttributeCheck( ( context, attributeName ) => {
	if ( context.endsWith( 'codeBlock $text' ) ) {
		return false;
	}
} );
```

Now it would be fired for, e.g. `schema.checkAttribute( [ 'image' ], 'imageSrc' )`, which doesn't make much sense. If we specify `$text` here, and only get callbacks for `context.last.name` and `attributeName`, then we could limit the number of callbacks even more.

Reply:
I guess making two properties to filter in both childCheck and attributeCheck seems reasonable. I can leave it like this to be improved, or try to improve it by this two-type filtering, tbd.